engine/classes/Elgg/EntityIconService.php

```diff
@@ -435,6 +435,11 @@ public function getSizes($entity_type = null, $entity_subtype = null, $type = 'i
 			$sizes = $this->hooks->trigger("entity:$type:sizes", $entity_type, $params, $sizes);
 		}
 
+		if (empty($sizes)) {
```

Comment:
That's my point. The only time you would have an empty $sizes is when the plugin is doing something silly.